### PR TITLE
Add flag to update the docs/ATT&CK-coverage.md with markdown URL(s)

### PR DESCRIPTION
**Pull request by shashank-elastic**
# Pull Request

*Issue link(s)*: https://github.com/elastic/detection-rules/issues/4007

## Summary - What I changed

- Add a flag `--update-coverage` to update the markdown links in `docs/ATT&CK-coverage.md`
- Currently I have used my GIST to test this, changes to `docs/ATT&CK-coverage.md` should be reverted before merging. 

To Discuss / Design

- We now build navigators on each push to main.
- How do we want to update the file, as this will be a commit and on push to main there will be new commits added and i sense this can cause issues and we dont want to be committing so frequently as well?
- Should we want to build off latest main regularly and update, Like a workflow to execute on Sunday 
- Should this be a prerequisite for a release process?  

## How To Test

- Currently locally we have run it against my GIST
- All the changes can be seen in `docs/ATT&CK-coverage.md` which should be reverted before merging! 
## Checklist

- [ ] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [ ] Secret and sensitive material has been managed correctly
- [ ] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Bug - Guidelines

These guidelines serve as a reminder set of considerations when addressing a bug in the code.

### Documentation and Context

- [ ] Provide detailed documentation (description, screenshots, reproducing the bug, etc.) of the bug if not already documented in an issue.
- [ ] Include additional context or details about the problem.
- [ ] Ensure the fix includes necessary updates to the release documentation and versioning.

### Code Standards and Practices

- [ ] Code follows established design patterns within the repo and avoids duplication.
- [ ] Code changes do not introduce new warnings or errors.
- [ ] Variables and functions are well-named and descriptive.
- [ ] Any unnecessary / commented-out code is removed.
- [ ] Ensure that the code is modular and reusable where applicable.
- [ ] Check for proper exception handling and messaging.

### Testing

- [ ] New unit tests have been added to cover the bug fix or edge cases.
- [ ] Existing unit tests have been updated to reflect the changes.
- [ ] Provide evidence of testing and detecting the bug fix (e.g., test logs, screenshots).
- [ ] Validate that any rules affected by the bug are correctly updated.
- [ ] Ensure that performance is not negatively impacted by the changes.
- [ ] Verify that any release artifacts are properly generated and tested.

### Additional Checks

- [ ] Ensure that the bug fix does not break existing functionality.
- [ ] Review the bug fix with a peer or team member for additional insights.
- [ ] Verify that the bug fix works across all relevant environments (e.g., different OS versions).
- [ ] Confirm that all dependencies are up-to-date and compatible with the changes.

**Comment by shashank-elastic** [comment]
Please note we need to decide the deployment model for this before merging, but the code can be reviewed

**Comment by shashank-elastic** [comment]
> Please note we need to decide the deployment model for this before merging, but the code can be reviewed

The deployment model is the workflow file to be run as part of the release process.

File: detection_rules/devtools.py [hunk 1]
```diff
@@ -820,17 +822,39 @@ def raw_permalink(raw_link):
...
+This also means they can be downloaded from all successful builds.
+
+These files can be used to pass to a custom navigator
+session. For convenience, the links are generated below. You can also include multiple across tabs in a single session,
+though it is not advisable to upload _all_ of them as it will likely overload your browsers resources.
+
+## Current rule coverage
+
+The source files for these links are regenerated with every successful merge to main. These represent coverage from the
+state of rules in the `main` branch."""
```
Comment (traut) [inline detection_rules/devtools.py:851]: to keep the formatting nice, we should use [`textwrap.dedent()`](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) here:

```python
    if update_coverage:
        # ...
        header_lines = textwrap.dedent("""
            # Rule coverage

            ATT&CK navigator layer files are generated when a package is built with
            `make release` or `python -m detection-rules`. This also means they can
            be downloaded from all successful builds. These files can be used to
            pass to a custom navigator session. For convenience, the links are
            generated below. You can also include multiple across tabs in a single
            session, though it is not advisable to upload _all_ of them as it will
            likely overload your browsers resources.

            ...
        """)
```

`textwrap.dedent()` will remove shared indent from all text lines, so the text will not have any margin on the left

**Review by traut — APPROVED** [review]

File: .github/workflows/attack-coverage-update.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip cache purge
+          pip install .[dev]
```
Comment (eric-forte-elastic) [inline .github/workflows/attack-coverage-update.yml:29]: Just double checking, might you need lib/kibana and lib/kql as well or perhaps not?

Comment (shashank-elastic) [reply .github/workflows/attack-coverage-update.yml:29]: I think we don't need them, we have mimed the workflow of .github/workflows/pythonpackage.yml where the artifacts are built and naviagtor gist is updated, which does not use these libs

Comment (eric-forte-elastic) [reply .github/workflows/attack-coverage-update.yml:29]: :thinking: That seems strange to me given that the `build-release` command is run as part of this workflow I think does require it.

Comment (shashank-elastic) [reply .github/workflows/attack-coverage-update.yml:29]: It is being installed as part of pip. Refer - https://github.com/elastic/detection-rules/blob/main/pyproject.toml#L41

Also verified from our workflow Job. 

image

We do not need an explicit install

Comment (eric-forte-elastic) [reply .github/workflows/attack-coverage-update.yml:29]: Ah its included in `.[dev]` now, so possibly no longer needed in the Makefile. 

<details><summary>Details</summary>
<p>

```bash
detection-rules on  main [$!?] is  v0.1.0 via  v3.12.6 (test_venv) on  eric.forte 
❯ pip install .[dev]
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing /home/forteea1/Code/clean_mains/detection-rules
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting detection-rules-kql@ git+https://github.com/elastic/detection-rules.git#subdirectory=lib/kql (from detection_rules==0.1.0)
  Cloning https://github.com/elastic/detection-rules.git to /tmp/pip-install-dcquu3e4/detection-rules-kql_d349e29b7d7c4178a81dd9eaae6635b8
  Running command git clone --filter=blob:none --quiet https://github.com/elastic/detection-rules.git /tmp/pip-install-dcquu3e4/detection-rules-kql_d349e29b7d7c4178a81dd9eaae6635b8
  Resolved https://github.com/elastic/detection-rules.git to commit 5e0fb4a63eb7a93f08b691a73ce3aa97a975fde6
  Running command git submodule update --init --recursive -q
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting detection-rules-kibana@ git+https://github.com/elastic/detection-rules.git#subdirectory=lib/kibana (from detection_rules==0.1.0)
  Cloning https://github.com/elastic/detection-rules.git to /tmp/pip-install-dcquu3e4/detection-rules-kibana_6ca2981e007a4fd0aa8a1c0cb209f3f9
  Running command git clone --filter=blob:none --quiet https://github.com/elastic/detection-rules.git /tmp/pip-install-dcquu3e4/detection-rules-kibana_6ca2981e007a4fd0aa8a1c0cb209f3f9
```

</p>
</details>

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:
